File: week_04/codewars/jwlee/tribonacci.java
public class Xbonacci {

  public double[] tribonacci(double[] s, int n) {

    double[] result = new double[n];
    
    for(int i = 0 ; i < n; i++) {
     result[i] = i<3? s[i]:result[i-1] + result[i-2] + result[i-3];
    }
    
    
     return result;

  }
}

File: week_04/codewars/cmSon/tribonacci.cpp
std::vector<int> tribonacci(std::vector<int> signature, int n)
{
    std::vector<int> result = signature;

    for (int i = 3; i < n; i++){
      int next_number = result[i - 3] + result[i - 2] + result[i - 1];
      result.push_back(next_number);
    }
    if (n < 3) result.resize(n);
    return result;
}